go/service/kvstore.go

```diff
@@ -237,3 +242,182 @@ func (h *KVStoreHandler) PutKVEntry(ctx context.Context, arg keybase1.PutKVEntry
 		Revision:  apiRes.Revision,
 	}, nil
 }
+
+func (h *KVStoreHandler) DelKVEntry(ctx context.Context, arg keybase1.DelKVEntryArg) (res keybase1.KVDeleteEntryResult, err error) {
+	ctx = libkb.WithLogTag(ctx, "KV")
+	mctx := libkb.NewMetaContext(ctx, h.G())
+	defer mctx.TraceTimed(fmt.Sprintf("KVStoreHandler#DeleteKVEntry: t:%s, n:%s, k:%s", arg.TeamName, arg.Namespace, arg.EntryKey), func() error { return err })()
+	if err := h.assertLoggedIn(ctx); err != nil {
+		mctx.Debug("not logged in err: %v", err)
+		return res, err
+	}
+	teamID, err := h.resolveTeam(mctx, arg.TeamName)
+	if err != nil {
+		mctx.Debug("error resolving team with name %s: %v", arg.TeamName, err)
+		return res, err
+	}
+	entryID := keybase1.KVEntryID{
+		TeamID:    teamID,
+		Namespace: arg.Namespace,
+		EntryKey:  arg.EntryKey,
+	}
+
+	// populate the local cache (this will make more sense after PICNIC-534)
+	getArg := keybase1.GetKVEntryArg(arg)
+	getRes, err := h.GetKVEntry(ctx, getArg)
+	if err != nil {
+		mctx.Debug("error populating the cache in team storage DEL flow: %+v", err)
+	}
+	revision := getRes.Revision + 1
+	////////////////////////////////
+
+	mctx.Debug("deleting %+v at revision %d", entryID, revision)
+	err = mctx.G().GetKVRevisionCache().CheckDeletable(mctx, entryID, revision)
+	if err != nil {
+		mctx.Debug("error comparing new entry %v against existing cache: %v", entryID, err)
+		return res, err
+	}
+	apiArg := libkb.APIArg{
+		Endpoint:    "team/storage",
+		SessionType: libkb.APISessionTypeREQUIRED,
+		Args: libkb.HTTPArgs{
+			"team_id":   libkb.S{Val: entryID.TeamID.String()},
+			"namespace": libkb.S{Val: entryID.Namespace},
+			"entry_key": libkb.S{Val: entryID.EntryKey},
+			"revision":  libkb.I{Val: revision},
+		},
+	}
+	apiRes, err := mctx.G().API.Delete(mctx, apiArg)
```

Comment:
I'm not a fan of using HTTP verbs here. These are API calls to a server, HTTP has nothing to offer but complexity. But I don't feel strongly.

Reply:
respectfully disagree. weakly.